daprdocs/content/en/developing-applications/develop-components/pluggable-components/develop-pluggable.md

```diff
@@ -105,6 +105,167 @@ After generating the above state store example's service scaffolding code using
+
+| Error                    | gRPC error code                 | 
+| ------------------------ | ------------------------------- | 
+| ETag Mismatch            | `codes.FailedPrecondition`          | 
+| ETag Invalid             | `codes.InvalidArgument`           | 
+| Bulk Delete Row Mismatch | `codes.Internal` | 
+
+Learn more about concurrency requirements in the [State Management overview]({{< ref "state-management-overview.md#concurrency" >}}).
+
+The following examples demonstrate how to return an error in your own pluggable component, changing the messages to suit your needs.
+
+{{< tabs ".NET" "Java" "Go" >}}
+ <!-- .NET -->
+{{% codetab %}}
+
+> **Important:** In order to use .NET for error mapping, first install the [`Google.Api.CommonProtos` NuGet package](https://www.nuget.org/packages/Google.Api.CommonProtos/).
+
+**Etag Mismatch**
+
+```csharp
+var badRequest = new BadRequest();
+var des = "The ETag field must only contain alphanumeric characters";
+badRequest.FieldViolations.Add(    
+   new Google.Rpc.BadRequest.Types.FieldViolation
+       {        
+         Field = "etag",
+         Description = des
+       });
+
+var baseStatusCode = Grpc.Core.StatusCode.FailedPrecondition;
+var status = new Google.Rpc.Status{    
+   Code = (int)baseStatusCode
+};
+
+status.Details.Add(Google.Protobuf.WellKnownTypes.Any.Pack(badRequest));
+
+var metadata = new Metadata();
+metadata.Add("grpc-status-details-bin", status.ToByteArray());
+throw new RpcException(new Grpc.Core.Status(baseStatusCode, "fake-err-msg"), metadata);
+```
+
+**Etag Invalid**
+
+```csharp
+var badRequest = new BadRequest();
+var des = "The ETag field must only contain alphanumeric characters";
+badRequest.FieldViolations.Add(
+   new Google.Rpc.BadRequest.Types.FieldViolation
+   {
+      Field = "etag",
+      Description = des
+   });
+
+var baseStatusCode = Grpc.Core.StatusCode.InvalidArgument;
+var status = new Google.Rpc.Status
+{
+   Code = (int)baseStatusCode
+};
+
+status.Details.Add(Google.Protobuf.WellKnownTypes.Any.Pack(badRequest));
+
+var metadata = new Metadata();
+metadata.Add("grpc-status-details-bin", status.ToByteArray());
+throw new RpcException(new Grpc.Core.Status(baseStatusCode, "fake-err-msg"), metadata);
+```
+
+**Bulk Delete Row Mismatch**
+
+```csharp
+var errorInfo = new Google.Rpc.ErrorInfo();
+
+errorInfo.Metadata.Add("expected", "100");
+errorInfo.Metadata.Add("affected", "99");
+
+var baseStatusCode = Grpc.Core.StatusCode.Internal;
+var status = new Google.Rpc.Status{
+    Code = (int)baseStatusCode
+};
+
+status.Details.Add(Google.Protobuf.WellKnownTypes.Any.Pack(errorInfo));
+
+var metadata = new Metadata();
+metadata.Add("grpc-status-details-bin", status.ToByteArray());
+throw new RpcException(new Grpc.Core.Status(baseStatusCode, "fake-err-msg"), metadata);
+```
+
+{{% /codetab %}}
+
+ <!-- Java -->
+{{% codetab %}}
+
+**ETag Mismatch**
+
+```java
+
+```
+
+**ETag Invalid**
+
+```java
+
+```
+
+**Bulk Delete Row Mismatch**
+
+```java
+
```

Comment:
```suggestion
    final Status status = Status.newBuilder()
        .setCode(io.grpc.Status.Code.INTERNAL.value())
        .setMessage("fake-err-msg-for-bulk-delete-row-mismatch")
        .addDetails(Any.pack(ErrorInfo.newBuilder()
            .putAllMetadata(Map.ofEntries(
                Map.entry("affected", "100"),
                Map.entry("expected", "99")
            ))
            .build()))
        .build();
    return Mono.error(StatusProto.toStatusException(status));
```

Reply:
Although I think that the values for affected and expected are swapped, I prefer to follow the example of the other code samples just for the sake of consistency. We might want to verify if that is correct and fix for the 3 languages.